Server/BackEndMonitoringServer/src/CServiceHandler.cpp

```diff
@@ -0,0 +1,287 @@
+#include "stdafx.h"
+
+#include "CServiceHandler.h"
+#include "Utils.h"
+
+class ServiceHandle
+{
+public:
+    explicit ServiceHandle(SC_HANDLE handle)
+      : m_handle(handle) {}
+
+    ~ServiceHandle() 
+    {
+        if (m_handle != nullptr) 
+        {
+            if(::CloseServiceHandle(m_handle) == 0)
+            {
+                Utils::DisplayError("Failed to close the service handle");
+            };
+        }
+    }
+
+    SC_HANDLE GetHandle() const
+    {
+        return m_handle;
+    }
+
+private:
+    SC_HANDLE m_handle = nullptr;
+};
+
+ServiceHandler::ServiceHandler(std::unique_ptr<CService> service)
+  : m_service(std::move(service))
+{ }
+
+bool ServiceHandler::Install() const
+{
+    Utils::DisplayMessage("Installing service");
+
+    bool success = true;
+
+    do
+    {
+        CString escaped_path;
+        LPSTR module_path = escaped_path.GetBufferSetLength(MAX_PATH);
+
+        // TODO: Wrap
+        if (::GetModuleFileName(nullptr, module_path, MAX_PATH) == 0)
+        {
+            Utils::DisplayError("Failed to get module file name");
+            success = false;
+            break;
+        }
+
+        escaped_path.ReleaseBuffer();
+        escaped_path.Remove('\"');
+
+        escaped_path = '\"' + escaped_path + '\"';
```

Comment:
maybe better write fucntion FormatePath(string)?

Reply:
Ok, @Serwlashev please, edit this in Sockets branch.